src/Controller/Entry/EntryFrontController.php

```diff
@@ -228,78 +146,93 @@ public function magazine(
             $response->headers->set('X-Robots-Tag', 'noindex, nofollow');
         }
 
-        $criteria = (new EntryPageView($this->getPageNb($request)));
+        if ('threads' === $content) {
+            $criteria = new EntryPageView($this->getPageNb($request));
+            $criteria->setContent('threads');
+        } elseif ('microblog' === $content) {
+            $criteria = new PostPageView($this->getPageNb($request));
+            $criteria->setContent('microblog');
+        } else {
+            throw new \LogicException('Invalid content '.$content);
+        }
+
         $criteria->showSortOption($criteria->resolveSort($sortBy))
-            ->setFederation(
-                'false' === $request->cookies->get(
-                    ThemeSettingsController::KBIN_FEDERATION_ENABLED,
-                    true
-                ) ? Criteria::AP_LOCAL : Criteria::AP_ALL
-            )
+            ->setFederation($federation)
             ->setTime($criteria->resolveTime($time))
             ->setType($criteria->resolveType($type));
+
+        $subscription = $request->query->get('subscription');
+
+        if ('sub' === $subscription) {
+            $this->denyAccessUnlessGranted('ROLE_USER');
+            $user = $this->getUserOrThrow();
+            $criteria->subscribed = true;
+        } elseif ('mod' === $subscription) {
+            $this->denyAccessUnlessGranted('ROLE_USER');
+            $criteria->moderated = true;
+        } elseif ('fav' === $subscription) {
+            $this->denyAccessUnlessGranted('ROLE_USER');
+            $criteria->favourite = true;
+        } elseif ($subscription && 'all' !== $subscription) {
+            throw new \LogicException('Invalid subscription filter '.$subscription);
+        }
+
         $criteria->magazine = $magazine;
         $criteria->stickiesFirst = true;
 
         if (null !== $user && 0 < \count($user->preferredLanguages)) {
             $criteria->languages = $user->preferredLanguages;
         }
 
-        $method = $criteria->resolveSort($sortBy);
-        $listing = $this->$method($criteria);
+        if ('threads' === $content) {
+            $listing = $this->entryRepository->findByCriteria($criteria);
+
+            $content_tmpl = 'entry/';
+            $content_key = 'entries';
+        } elseif ('microblog' === $content) {
+            $listing = $this->postRepository->findByCriteria($criteria);
+
+            $content_tmpl = 'post/';
+            $content_key = 'posts';
+        }
```

Comment:
We already had a check for that up above nearer the beginning of the function, but sure I can add another one here for safety in case something changes / something weird happens.